src/CommerceInsights.php

```diff
@@ -19,58 +25,52 @@
-
-                /*
-                 * Products
+                 * Vue templates
                  */
-                $event->rules['commerceinsights/products']               = 'commerceinsights/product/product-index';
-                $event->rules['commerceinsights/products/<format:json>'] = 'commerceinsights/product/product-index';
-                $event->rules['commerceinsights/products/<format:csv>']  = 'commerceinsights/product/product-index';
-
-                /*
-                 * Sales
-                 */
-                $event->rules['commerceinsights/sales']               = 'commerceinsights/sales/sales-index';
-                $event->rules['commerceinsights/sales/<format:json>'] = 'commerceinsights/sales/sales-index';
-                $event->rules['commerceinsights/sales/<format:csv>']  = 'commerceinsights/sales/sales-index';
-
-                /*
-                 * Customers
-                 */
-                $event->rules['commerceinsights/customers']               = 'commerceinsights/customer/customer-index';
-                $event->rules['commerceinsights/customers/<format:json>'] = 'commerceinsights/customer/customer-index';
-                $event->rules['commerceinsights/customers/<format:csv>']  = 'commerceinsights/customer/customer-index';
+                $event->rules['commerceinsights/view/orders']                  = 'commerceinsights/orders/index';
+                $event->rules['commerceinsights/orders/product/<id:([0-9])+>'] = 'commerceinsights/product/index';
+                $event->rules['commerceinsights/view/items-sold']              = 'commerceinsights/items-sold/index';
+                $event->rules['commerceinsights/view/customers']               = 'commerceinsights/customers/index';
+            }
+        );
 
+        Event::on(UrlManager::class, UrlManager::EVENT_REGISTER_SITE_URL_RULES,
+            function (RegisterUrlRulesEvent $event) {
                 /*
-                 * Saved
+                 * AJAX routes
                  */
-                $event->rules['commerceinsights/saved']             = 'commerceinsights/saved/list';
-                $event->rules['commerceinsights/saved/save']        = 'commerceinsights/saved/save-report';
-                $event->rules['commerceinsights/saved/<id>']        = 'commerceinsights/saved/get-report';
-                $event->rules['commerceinsights/saved/delete/<id>'] = 'commerceinsights/saved/delete-report';
-
-                // new vue stuff
-                $event->rules['commerceinsights/view/<view>'] = 'commerceinsights/vue/index';
-            });
+                $event->rules['get-ci-orders']     = 'commerceinsights/orders/get-orders';
+                $event->rules['get-ci-items-sold'] = 'commerceinsights/items-sold/get-items-sold';
+                $event->rules['get-ci-product']    = 'commerceinsights/product/get-product';
+                $event->rules['get-ci-customers']  = 'commerceinsights/customers/get-customers';
```

Comment:
I can probably put the handle in the routes (maybe), but if I let Craft generate the action routes, what ends up happening is that we can't let other plugins override the routes to introduce their own methods. For example, for Chrome Emblems, we need a completely different method for how "Items Sold" is displayed (we need to re-reoute to a custom method in a separate plugin for Chrome Emblems). If we use the default action routes, for example `/actions/commerceinsights/items-sold/get-items-sold`, it can't be overridden (I tried), so the method in my Chrome Emblems plugin won't be able to run at that URI.